Create a registration UI for agents

GuiBibeau opened the pull request:
## Description

Allows registering agents using a graphical interface

## Testing

1. run this repo locally, add a private key of a wallet with sepolia base eth
2. navigate to /register
3. perform a registration
4. notice success screen


https://github.com/user-attachments/assets/60895214-7cd6-4d95-8ac3-b580ee398604



## Screenshots
<img width="734" alt="Screenshot 2025-02-05 at 8 04 18 AM" src="https://github.com/user-attachments/assets/95722d71-4445-46ef-924a-62669e8092dd" />

<img width="789" alt="Screenshot 2025-02-05 at 8 00 41 AM" src="https://github.com/user-attachments/assets/9a0e8026-11e1-429e-80a6-0f16a9ccd7fe" />

GuiBibeau committed:
add shadcn ui

GuiBibeau committed:
Merge remote-tracking branch 'origin/main' into registration-ui

GuiBibeau committed:
bring in shadcn components for UI

GuiBibeau committed:
remove optional tag on twitter schema to let schema consumers optionally use it

GuiBibeau committed:
create server action to submit agent

GuiBibeau committed:
create page and form

GuiBibeau committed:
create success page

GuiBibeau committed:
visually show required fields

GuiBibeau committed:
merge main

GuiBibeau committed:
remove test that is no longer necessary

codenamejason commented:
Should we add an `env.example` file?

codenamejason reviewed (approved):
Was able to register running locally.

GuiBibeau committed:
feat: better starting instructions

GuiBibeau commented:
> Should we add an `env.example` file?

Great point! I updated the readme and added an example `.env` with better description. Now using this should be easier
